lib/src/common_server_api.dart

```diff
@@ -107,7 +107,10 @@ class CommonServerApi {
           decodeFromProto: (bytes) => proto.VersionRequest.fromBuffer(bytes),
           transform: _impl.version);
 
-  Router get router => _$CommonServerApiRouter(this);
+  Router? _router;
+
+  /// Return the router, creating it if we have not done so already.
+  Router get router => (_router ??= _$CommonServerApiRouter(this));
```

Comment:
I'm not sure if there was a reason it was constructed each time before, but you can use `late final` to make this type of  structure a bit easier to read:

```suggestion
  /// The (lazily-constructed) router.
  late final Router router = _$CommonServerApiRouter(this);
```

Reply:
Yes - that is much better.  I have corrected that.

The existing code accesses the getter only once, when the server is being created in `services_cloud_run.dart`  or `services_dev.dart`.

Previously there was no occasion to access it again.  It seems to me that if the previous behavior was desired then this should be a method call with an appropriate name and not a getter.